src/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php

```diff
@@ -126,6 +127,12 @@ public function processNodes(
                 return;
             }
 
+            if ($node instanceof StmtsAwareInterface && $node->stmts !== null) {
+                foreach ($node->stmts as $stmt) {
+                    $stmt->setAttribute(AttributeKey::SCOPE, $mutatingScope);
```

Comment:
Using `$this->nodeScopeResolver->processNodes($node->stmts, $mutatingScope, $nodeCallback);` may probably better for reliable enter scope.

I will check if that may cause some performance hit.

Reply:
Testing manually with multi stmts, the scope seems linear, so it seems ok (for now), note to myself: it may need to revisit.

Reply:
https://github.com/rectorphp/rector-src/pull/5450